src/Avalonia.Base/Threading/Dispatcher.Invoke.cs

```diff
@@ -625,6 +625,35 @@ public void Post(SendOrPostCallback action, object? arg, DispatcherPriority prio
         _ = action ?? throw new ArgumentNullException(nameof(action));
         InvokeAsyncImpl(new SendOrPostCallbackDispatcherOperation(this, priority, action, arg, true), CancellationToken.None);
     }
+    
+    /// <summary>
+    /// Sends an action that will be invoked on the dispatcher thread.
+    /// </summary>
+    /// <param name="action">The method.</param>
+    /// <param name="arg">The argument of method to call.</param>
+    /// <param name="priority">The priority with which to invoke the method.</param>
+    internal void Send(SendOrPostCallback action, object? arg, DispatcherPriority priority)
+    {
+        _ = action ?? throw new ArgumentNullException(nameof(action));
+        if (priority == DispatcherPriority.Send && CheckAccess())
+        {
+            try
+            {
+                using (AvaloniaSynchronizationContext.Ensure(this, priority))
+                    action(arg);
+            }
+            catch (Exception ex) when (ExceptionFilter(ex))
+            {
+                if (!CatchException(ex))
+                    throw;
+            }
+        }
+        else
+        {
+            InvokeAsyncImpl(new SendOrPostCallbackDispatcherOperation(this, priority, action, arg, true),
```

Comment:
Send implies a synchronous wait, InvokeAsyncImpl won't wait for the operation to be completed.

Reply:
Fixed.